src/main/java/com/ldtteam/blockout/controls/AbstractTextElement.java

```diff
@@ -1,101 +1,392 @@
+            renderedTextWidth = (int) (preparedText.stream().mapToInt(mc.fontRenderer::func_243245_a).max().orElse(maxWidth) * textScale);
+            renderedTextHeight = (int) (Math.min(preparedText.size() * this.mc.fontRenderer.FONT_HEIGHT, maxHeight) * textScale);
+        }
+        else
+        {
+            preparedText = preparedText.subList(0, 1);
+            renderedTextWidth = (int) (mc.fontRenderer.func_243245_a(preparedText.get(0)) * textScale);
+            renderedTextHeight = (int) (this.mc.fontRenderer.FONT_HEIGHT * textScale);
+        }
+    }
+
+    protected int getTextRenderingColor(final double mx, final double my)
+    {
+        return isPointInPane(mx, my) ? textHoverColor : textColor;
+    }
+
+    @Override
+    public void drawSelf(final MatrixStack ms, final double mx, final double my)
+    {
+        if (preparedText.isEmpty())
+        {
+            return;
+        }
+
+        final int color = enabled ? (isPointInPane(mx, my) ? textHoverColor : textColor) : textDisabledColor;
+
+        int offsetX = textOffsetX;
+        int offsetY = textOffsetY;
+
+        if (textAlignment.isRightAligned())
+        {
+            offsetX += textWidth - renderedTextWidth;
+        }
+        else if (textAlignment.isHorizontalCentered())
+        {
+            offsetX += (textWidth - renderedTextWidth) / 2;
+        }
+
+        if (textAlignment.isBottomAligned())
+        {
+            offsetY += textHeight - renderedTextHeight;
+            offsetY += Math.ceil(textScale);
+        }
+        else if (textAlignment.isVerticalCentered())
+        {
+            offsetY += (textHeight - renderedTextHeight) / 2;
+            offsetY += Math.ceil(textScale);
+        }
+
+        ms.push();
+        ms.translate(x + offsetX, y + offsetY, 0.0d);
+        ms.scale((float) textScale, (float) textScale, 1.0f);
+
+        final Matrix4f matrix4f = ms.getLast().getMatrix();
+        int lineShift = 0;
+        for (final IReorderingProcessor row : preparedText)
+        {
+            final int xOffset;
+
+            if (textAlignment.isRightAligned())
+            {
+                xOffset = (int) ((renderedTextWidth - mc.fontRenderer.func_243245_a(row) * textScale) / textScale);
+            }
+            else if (textAlignment.isHorizontalCentered())
+            {
+                xOffset = (int) ((renderedTextWidth - mc.fontRenderer.func_243245_a(row) * textScale) / 2 / textScale);
+            }
+            else
+            {
+                xOffset = 0;
+            }
+
+            mc.fontRenderer.func_238415_a_(row, xOffset, lineShift, color, matrix4f, textShadow);
+            lineShift += mc.fontRenderer.FONT_HEIGHT + textLinespace;
+        }
+
+        ms.pop();
+    }
+
+    public Alignment getTextAlignment()
+    {
+        return textAlignment;
+    }
+
+    public void setTextAlignment(final Alignment textAlignment)
```

Comment:
missing javadoc

Reply:
a bunch of methods in this class

Reply:
Do we really need jdoc for set/get methods?

Reply:
@ShadowProtocol what do you think of this?

Reply:
Same probably applies to the text builder

Reply:
I just imagine it as if it is in an auto-genned doc page. would someone understand that this sets the text alignment based off of the method title alone? If the answer is yes then it doesn't need a jdoc IMO. You just need to be careful that you aren't doing other things in the setter that won't be described by the title alone. 
But if you're just going to chuck ``/** Sets the text alignment */`` above it then that is not worth anyone's energy scrolling past it.